ComponentTextKit/TextKit/CKTextKitAttributes.h

```diff
@@ -77,6 +77,10 @@ struct CKTextKitAttributes {
    The radius that should be applied to the shadow blur.  Larger values mean a larger, more blurred shadow.
    */
   CGFloat shadowRadius;
+  /**
+   A pointer to a function that that returns a custom layout manager subclass. If nil, defaults to NSLayoutManager.
+   */
+  NSLayoutManager *(*layoutManagerFactory)(void);
```

Comment:
Switched to the function pointer approach as you suggested. This is nice since we get some type safety from the compiler and no need for internal assertions.